src/makeCalculator.js

```diff
@@ -4,7 +4,48 @@
  * @return {object}
  */
 function makeCalculator() {
-  // write code here
+  const calculator = {
+    result: 0,
+    add(number) {
+      this.result += number;
+
+      return this;
+    },
+
+    subtract(number) {
+      this.result -= number;
+
+      return this;
+    },
+
+    multiply(number) {
+      this.result *= number;
+
+      return this;
+    },
+
+    divide(number) {
+      if (number !== 0) {
+        this.result /= number;
+      }
+
+      return this;
+    },
+
+    reset() {
+      this.result = 0;
+
+      return this;
+    },
+
+    operate(callback, number) {
+      callback.call(this, number);
+
+      return this;
+    },
+  };
+
+  return calculator;
 }
```

Comment:
The structure of the `makeCalculator` function is correct, and it is following the task requirements. However, consider checking for division by zero inside the `operate` method instead of only in the `divide` method. This would centralize the error checking logic and make the `divide` method cleaner.